src/main/kotlin/org/jitsi/jicofo/conference/source/EndpointSourceSet.kt

```diff
@@ -41,6 +41,16 @@ data class EndpointSourceSet(
      */
     fun isEmpty() = sources.isEmpty() && ssrcGroups.isEmpty()
 
+    /**
+     * Whether there are any audio sources in this set.
+     */
+    fun hasAudio() = sources.any { it.mediaType == AUDIO }
```

Comment:
Nit since you're optimizing anyway :)
`EndpointSourceSet` is immutable so this could be a `val` calculated once. And if you use the `lazy` delegate it will only be initialized on demand: see e.g. the definition of `audioSsrcs` below.

Reply:
cool, thanks.